transform.go

```diff
@@ -424,12 +426,18 @@ func (t *Transformer) transformAnyEveryIn(expr AnyEveryInExpr) *ExecNode {
 	return t.transformLoop(LoopTypeAnyEvery, expr.VarId, expr.InExpr, expr.SubExpr)
 }
 
-func (t *Transformer) makeRhsParam(expr Expression) interface{} {
+func (t *Transformer) makeRhsParam(expr Expression, op OpType) interface{} {
 	if rhsField, ok := expr.(FieldExpr); ok {
 		rhsNode := t.getExecNode(rhsField)
 		rhsStoreId := rhsNode.makeStored(t)
 		return VarRef{rhsStoreId}
 	} else if rhsValue, ok := expr.(ValueExpr); ok {
+		if op == OpTypeMatches {
```

Comment:
Parameter parsing should never be op-specific, but instead be more explicit through the parameter.

Reply:
Done